themes-default/slim/src/components/edit-show.vue

```diff
@@ -0,0 +1,437 @@
+        showIndexer: {
+            type: String
+        },
+        /**
+         * Show id
+         */
+        showId: {
+            type: Number
+        }
+    },
+    data() {
+        return {
+            saving: false
+        };
+    },
+    computed: {
+        ...mapState({
+            config: state => state.config,
+            defaultShow: state => state.defaults.show,
+            episodeStatuses: state => state.consts.statuses
+        }),
+        ...mapGetters([
+            'getShowById',
+            'getStatus'
+        ]),
+        indexer() {
+            return this.showIndexer || this.$route.query.indexername;
+        },
+        id() {
+            return this.showId || Number(this.$route.query.seriesid) || undefined;
+        },
+        show() {
+            const { defaultShow, getShowById, id, indexer } = this;
+            return getShowById({ indexer, id }) || defaultShow;
+        },
+        defaultEpisodeStatusOptions() {
+            if (this.episodeStatuses.length === 0) {
+                return [];
+            }
+            // Get status objects, in this order
+            return ['wanted', 'skipped', 'ignored'].map(key => this.getStatus({ key }));
+        },
+        availableLanguages() {
+            if (this.config.indexers.config.main.validLanguages) {
+                return this.config.indexers.config.main.validLanguages.join(',');
+            }
+
+            return '';
+        },
+        combinedQualities() {
+            const { allowed, preferred } = this.show.config.qualities;
+            return combineQualities(allowed, preferred);
+        },
+        saveButton() {
+            return this.saving === false ? 'Save Changes' : 'Saving...';
+        },
+        globalIgnored() {
+            return this.$store.state.search.filters.ignored.map(x => x.toLowerCase());
+        },
+        globalRequired() {
+            return this.$store.state.search.filters.required.map(x => x.toLowerCase());
+        },
+        effectiveIgnored() {
+            const { globalIgnored } = this;
+            const showIgnored = this.show.config.release.ignoredWords.map(x => x.toLowerCase());
+
+            if (this.show.config.release.ignoredWordsExclude) {
+                return arrayExclude(globalIgnored, showIgnored);
+            }
+
+            return arrayUnique(globalIgnored.concat(showIgnored));
+        },
+        effectiveRequired() {
+            const { globalRequired } = this;
+            const showRequired = this.show.config.release.requiredWords.map(x => x.toLowerCase());
+
+            if (this.show.config.release.requiredWordsExclude) {
+                return arrayExclude(globalRequired, showRequired);
+            }
+
+            return arrayUnique(globalRequired.concat(showRequired));
+        }
+    },
+    created() {
+        const { getShows } = this;
+        // Without getting any specific show data, we pick the show needed from the shows array.
+        getShows();
```

Comment:
@p0psicles Are we getting all the shows in order to show the `ShowSelector`?
It's currently not visible [because it's limited to `displayShow` only:

https://github.com/pymedusa/Medusa/blob/63a6b28789bebb3f8621bfecf816d5bedfe61455/themes-default/slim/src/components/sub-menu.vue#L43-L46

Reply:
No we're getting the shows just for that one show we want to edit. So maybe we shouldn't do that

Reply:
Okay, although I'm keep having an reproducible issue - the show is not available at render time because we're fetching from an API.
I will have to continue this tomorrow... but I'll probably go with a `v-if` and a loading spinner.

Reply:
Mm how is that different from getting all shows?

Reply:
Couldn't let this go and turns out it was just bad `QualityChooser` design on my part...
But to answer your question, I think it's just a simple race condition.